app/services/data_decoder.py

```diff
@@ -431,6 +441,36 @@ async def decode_transaction(
         }
         return fn_name, decoded_transactions
 
+    async def get_decoding_accuracy(
+        self,
+        data: bytes | str,
+        address: Address | None = None,
+        chain_id: int | None = None,
+    ) -> DecodingAccuracyEnum:
+        """
+        Get decoding accuracy:
+            - FULL_MATCH: Contract `address` and `chain_id` matching
+            - PARTIAL_MATCH: Only contract `address` matching
+            - ONLY_FUNCTION_MATCH: Match with a function of another contract
+            - NO_MATCH: Cannot decode `data`
+
+        :param data:
+        :param address:
+        :param chain_id:
+        :return: DecodingAccuracyEnum
+        """
+        selector = HexBytes(data)[:4]
+        if selector not in self.fn_selectors_with_abis:
+            return DecodingAccuracyEnum.NO_MATCH
+        if address is not None:
+            if chain_id is not None and await self.get_contract_abi(
+                address, chain_id=chain_id
```

Comment:
get_contract_abi is being called previously, wouldn´t be possible decide the partil match or full match there? 
This would make a second query to the database just to get the match level without uses the abi.

Reply:
This is my reasoning:
  1. There's no database involved. Everything is cached due to the first call to the decoder
  2. Doing what you are proposing implies changing almost all the return methods for the whole service, to return 
      a information that's not directly related to the information returned. Or maybe I'm missing how to do it better, 
      do you have a proposal

Reply:
If is cached is right.